### ISSUE TITLE
Not Functional in TAO 3.1.0 RC7?

### ISSUE THREAD

**Issue by CodeMouse92** [oat-sa/generis-auth-ldap#3]
I followed the setup instructions, which seemed to work fine, and then I edit `/path/to/tao/config/generis/auth.conf.php` so it now reads:

```
return array(
 16     array(
 17         'driver' => 'oat\\authLdap\\model\\LdapAdapter',
 18         'config' => array(
 19             array(
 20                 'host' => '127.0.0.1',
 21                 'accountDomainName' => 'mycompany.com',
 22                 'baseDn' => 'ou=Users, dc=ldap, dc=mycompany, dc=com',
 23                 'bindRequiresDn' => 'true',
 24             )
 25         )
 26     ),
 27     array(
 28         'driver' => 'oat\\generis\\model\\user\\AuthAdapter',
 29         'hash' => array(
 30             'algorithm' => 'sha256',
 31             'salt' => 10
 32         )
 33     )
 34 );
```

This should connect to the same localhost LDAP server that everything else uses successfully. (It does not require login to check names). However, when I try to log in to TAO using any of the LDAP credentials that work elsewhere, it says the username/password is wrong.

``cn`` is the username, while ``mail`` is the email address.

How do I fix this?

(Ubuntu 16.04 LAMP server. PHP 5)

**Comment by jbout** [comment]
Could you please enable your local logger (see https://github.com/oat-sa/taohub-articles/blob/master/forge/Documentation%20for%20core%20components/logger.md) to see what exactly goes wrong?

**Comment by cigzigwon** [comment]
I'm confirming that it works in that version all except I can't get the LDAP test takers to see any tests. You might want to try setting 'bindRequiresDn' to 'false' and also you probably don't have an ldap server on your localhost do you?

**Comment by cigzigwon** [comment]
So the new issue is that LDAP users can indeed login but there's no way to set them to a group so that they can get the test releases. Maybe I'm doing it wrong? None of this documentation is very clear in this plugin.
<pre>
'mapping' => array(
      'http://www.tao.lu/Ontologies/TAOGroup.rdf#Member' => array(
        'type' => 'value',
        'value' => '#ldap',
      ),
    ),
</pre>

And I have created the LDAP group and released the demo test. I see nothing from my LDAP account.

**Comment by cigzigwon** [comment]
This all works based on... https://forge.taotesting.com/boards/6/topics/2345
But after taking the test the test taker cannot be identified. I'm going to try some hacks but in this plugin I don't see first, last, or account name being applied.

**Comment by cigzigwon** [comment]
Based on what you have mentioned in your blog you may as well scrap the way this works. Companies with LDAP want a syncing feature into the TAO DB so we can also set roles and things of that sort. I don't see why you guys just didn't start there...

**Comment by jbout** [comment]
While we hope to have a full user abstraction in the future, currently only the authentication and starting a user session via LDAP are supported, because we were able to do this rather quickly and there was not a lot of demand for a deeper LDAP integration.

In general we recommend to have a single source of truth for each resource, but if you really prefer to sync resources from LDAP to Tao, then you can do this using a custom script and the REST interfaces already. This would allow you to assign the users you have imported to groups using the Tao user interface, but there might be a few edge-cases such as users that have been deleted, that you will need to take into consideration.

**Comment by cigzigwon** [comment]
My advice would just be to scrap the work on the LDAP plugin because realistically it's useless. No instructor wants to see test results with no names. And no developer wants to waste time hacking at the RESTful interface when there is nothing in the interface specifically for this problem. I'm required to send two params that I don't have and then setup auth basic which I don't want to use. It's too much work for what it is when I can just build in my own support for LDAP syncing as I am a software engineer.

I recommend everyone to avoid wasting their time with this LDAP plugin.

**Comment by jbout** [comment]
I agree that there are not many use-cases for using this experimental plugin. If you are going to work in an alternative approach that could be beneficial to many, please let me know so that we can support you.

Alternatively take a look at the LTI interface https://www.imsglobal.org/activity/learning-tools-interoperability, that we use for most of our integrations these days: https://github.com/oat-sa/extension-tao-ltideliveryprovider

**Comment by cigzigwon** [comment]
I didn't plan on integrating it directly but if you would like me to. I'm sure I can help in this development. This is still a great product! Thanks for all of your advice up until now.
